firebase-config/src/test/java/com/google/firebase/remoteconfig/internal/ConfigContainerTest.java

```diff
@@ -336,6 +336,120 @@ public void configContainer_withRolloutsMetadata_returnsCorrectRolloutsMetadata(
         .isEqualTo(rolloutsMetadata.toString());
   }
 
+  @Test
+  public void getChangedParams_changedRolloutMetadata_returnsUpdatedKey() throws Exception {
+    JSONArray activeRolloutMetadata = generateRolloutMetadata(1);
+    JSONArray fetchedRolloutMetadata = generateRolloutMetadata(1);
+
+    fetchedRolloutMetadata.getJSONObject(0).put("variantId", "B");
+
+    ConfigContainer config =
+        ConfigContainer.newBuilder()
+            .replaceConfigsWith(ImmutableMap.of("key_1", "value_1"))
+            .withRolloutMetadata(activeRolloutMetadata)
+            .build();
+
+    ConfigContainer other =
+        ConfigContainer.newBuilder()
+            .replaceConfigsWith(ImmutableMap.of("key_1", "value_1"))
+            .withRolloutMetadata(fetchedRolloutMetadata)
+            .build();
+
+    Set<String> changedParams = config.getChangedParams(other);
+
+    assertThat(changedParams).containsExactly("key_1");
+  }
+
+  @Test
+  public void getChangedParams_addedRolloutMetadataToSameKey_returnsUpdatedKey() throws Exception {
+    JSONArray activeRolloutMetadata = generateRolloutMetadata(1);
+    JSONArray fetchedRolloutMetadata = generateRolloutMetadata(2);
+
+    ConfigContainer config =
+        ConfigContainer.newBuilder()
+            .replaceConfigsWith(ImmutableMap.of("key_1", "value_1"))
+            .withRolloutMetadata(activeRolloutMetadata)
+            .build();
+
+    ConfigContainer other =
+        ConfigContainer.newBuilder()
+            .replaceConfigsWith(ImmutableMap.of("key_1", "value_1"))
+            .withRolloutMetadata(fetchedRolloutMetadata)
+            .build();
+
+    Set<String> changedParams = config.getChangedParams(other);
+
+    assertThat(changedParams).containsExactly("key_1");
+  }
+
+  @Test
+  public void getChangedParams_deletedRolloutMetadata_returnsUpdatedKey() throws Exception {
+    JSONArray activeRolloutMetadata = generateRolloutMetadata(1);
+    JSONArray fetchedRolloutMetadata = new JSONArray();
+
+    ConfigContainer config =
+        ConfigContainer.newBuilder()
+            .replaceConfigsWith(ImmutableMap.of("key_1", "value_1"))
+            .withRolloutMetadata(activeRolloutMetadata)
+            .build();
+
+    ConfigContainer other =
+        ConfigContainer.newBuilder()
+            .replaceConfigsWith(ImmutableMap.of("key_1", "value_1"))
+            .withRolloutMetadata(fetchedRolloutMetadata)
+            .build();
+
+    Set<String> changedParams = config.getChangedParams(other);
+
+    assertThat(changedParams).containsExactly("key_1");
+  }
+
+  @Test
+  public void getChangedParams_addNewRolloutMetadata_returnsUpdatedKey() throws Exception {
```

Comment:
```suggestion
  public void getChangedParams_addNewKeyWithRolloutMetadata_returnsNewKey() throws Exception {
```

does that describe the test case better?